### ISSUE TITLE
Wrong precision visible for numeric datatype (result grid) with Postgres

### ISSUE THREAD

**Issue by andywicht** [dbeaver/dbeaver#5077]
#### System information: 
- Operating system Win 10
- DBeaver version 5.3.3
- Java version 

#### Description
When using the grid view for results, numeric datatypes are truncated for display. When double clicking into the result cell, it shows the correct precision.
Text result view also shows the correct precision.

This can lead to user confusion ;) Not sure if locale settings also play a role here.

#### Steps to reproduce:
`SELECT 123.4::numeric(9,6);`

![grid](https://user-images.githubusercontent.com/21099522/51601716-6b7a8580-1f05-11e9-81f0-828166884651.png)

![text](https://user-images.githubusercontent.com/21099522/51601718-6cabb280-1f05-11e9-9852-920f42bb0c65.png)

**Comment by serge-rider** [comment]
This is configurable in preferences (global or connection-specific). 
You can change minimum fraction digits: 
![image](https://user-images.githubusercontent.com/6398845/51983668-5c1dae00-24aa-11e9-8986-af439fce77b9.png)

In plaintext mode DBeaver uses native numbers representation by default.

**Comment by andywicht** [comment]
Hmm, applying the settings shown in the screenshot does not change the result if it is zeros at the end.

Compare those queries:
`SELECT 123.000000::numeric(9,6);` Displayed as: 123.0000
`SELECT 123.999999::numeric(9,6);` Displayed as: 123.999999

**Comment by serge-rider** [comment]
Yep. I've found the reason. We limit numeric data type scale to 4 by default. I can't remember why 4, looks like a bug. I'll set it to 16 (because some databases return wrong huge numeric scale, like thousands of fraction digits, we can't trust it).

**Comment by tobwen** [comment]
Sorry, seems like you'll have to reopen this. I'm on v6.2.2.201910061950 on Windows (running PostgreSQL on Debian).

Even with activated _Use data type scale for fraction digits_, `SELECT 1/3::NUMERIC(10,9) AS test;` gets clipped to three digits in grid view.

![image](https://user-images.githubusercontent.com/1864057/67150654-95a57c00-f2ba-11e9-8415-c0dff7526645.png)

The database correctly return
```
test                  |
----------------------|
0.33333333333333333333|
```

**Comment by serge-rider** [comment]
We've added new parameter for numbers formatting:
![image](https://user-images.githubusercontent.com/6398845/67759401-6c21e880-fa50-11e9-9cee-4d888f71b130.png)

DBeaver will respect database column data type scale and precision is this option is enabled.

It is disabled by default as for many databases it doesn't make sense, e.g. scale for all numbers = 255 by default or other crazy things.

**Comment by tobwen** [comment]
@serge-rider I know, but it doesn't seem to work here.

![image](https://user-images.githubusercontent.com/1864057/67801807-40bbef80-fa8a-11e9-95f0-44b9cf3267fa.png)

**Comment by serge-rider** [comment]
@tobwen query expressions don't have explicit scale in PostgreSQL. This will work only for real table columns or constant expressions with explicit type. Or jdbc driver doesn't support it - I'm not sure.
For example `SELECT 123.000000::numeric(9,3)` works ok.

What result do you expect for query `SELECT 1/3::numeric`?

**Comment by tobwen** [comment]
`SELECT 123.000000::numeric(9,3)` of course only returns 3 decimals, `SELECT 123.000000::numeric(9,4)` gives me 4. But `SELECT 1/3::numeric(9,4)` only gives 3.

JDBC driver seems to return the correct result: When you double click it, it shows all the missing digits:

![image](https://user-images.githubusercontent.com/1864057/67803109-a7daa380-fa8c-11e9-94a7-170aa32be7a0.png)

Expected is the same precision as _psql_: 
```
=# SELECT 1/3::numeric;
┌────────────────────────┐
│        ?column?        │
├────────────────────────┤
│ 0.33333333333333333333 │
└────────────────────────┘
(1 row)
```

JDBC correctly transmits this value to DBeaver, but the GUI seems to clip the value.

**Comment by serge-rider** [comment]
@tobwen I can't reproduce this. Please make sure that your connection-specific settings for numbers formatting don't set "Maximum fraction digits" to 3:

![image](https://user-images.githubusercontent.com/6398845/67874818-8e962d80-fb46-11e9-8b3d-ed2864f60ff6.png)

**Comment by tobwen** [comment]
@serge-rider As you can see in my screenshot, it hasn't been fixed.

Interesting, you're closing this issue although the error still exists... Nevermind.

**Comment by serge-rider** [comment]
@tobwen reopening an issue is just one click - standard part of the process.

Please write a bit more info:
- Did you check it in the Early Access version?
- Are you sure your connection-specific preferences are correct?

**Comment by Diabolarius** [comment]
@serge-rider I dicovered this ticket as the problem is still persistent. Therefore please reopen the ticket. 
As can be seen in the screenshot the fraction-digits of the preference doesnt affect the GUI-view. This view is instead always limited to 3 fraction digits max and half-rounding is applied. Nonethless of the preferences in data formats. 
Honestly, I cannot understand what @tobwen or myself need to provide more info. You can easily recreate this issue by storing an example value in an example DB (for example 0.3434545) and look at it with DBeaver. It will nonethless of any pref. always be rounded to 3 fraction digits.
The wanted behavior is a possibility to see the actual value stored in the DB with the fraction-digits configured in some preferences in the GUI-overview of the table without clicking in each cell to see the actual value. 
![grafik](https://user-images.githubusercontent.com/62017641/148547433-5b3dc0d0-6660-4a99-9b0a-a75d692eb124.png)
![grafik](https://user-images.githubusercontent.com/62017641/148547585-8c9607da-8749-497c-9ec8-acec2e3154ef.png)

**Comment by jianzhaofeng** [comment]
I have the latest version and  got same issue, please reopen this issue.
![image](https://user-images.githubusercontent.com/17436601/151386570-2a85f948-10ba-420d-a46a-2d4d52edd756.png)
![image](https://user-images.githubusercontent.com/17436601/151386849-6f7d5857-9d93-422e-a005-2e1866794507.png)

**Comment by uslss** [comment]
@jianzhaofeng latest version is 21.3.3
you could download it and check if issue is still actual https://dbeaver.io/download/

**Comment by LonwoLonwo** [comment]
Hello @Diabolarius 

Your Table DDL will be beneficial.

**Comment by jianzhaofeng** [comment]
> @jianzhaofeng latest version is 21.3.3 you could download it and check if issue is still actual https://dbeaver.io/download/

Thanks a lot, It has been fix at  Version 21.3.3

**Comment by matthung0807** [comment]
Mac version, go to [Preferences] > [Editors] > [Data Editor] > [Data Formats], check [Disable numeric formatting] in [Settings] will display right digits precision.

![截圖 2023-08-30 上午11 22 31](https://github.com/dbeaver/dbeaver/assets/20328567/567a6cb6-9976-456d-b688-a52296b40211)
